### PR TITLE
Add group selection to developer topology view

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Implements https://jira.coreos.com/browse/ODC-1796

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/TopologyResourcePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+/**
+ * REMOVE: once we get labels in place
+ * This is a temporary check to avoid the `Warning: Each child in an array or iterator should have a unique "key" prop`.
+ * Its coming when buildConfig/route/service metadata is empty object,
+ * BuildOverviewList, RouteOverviewList, ServiceOverview list uses the metadata.uid as the value of `key` prop.
+ *
+ * Datacontroller get the buildConfigs based on apps.kubernetes.io/instance label which is not applied to apps created using browser catalog
+ */
+
+function metadataUIDCheck(items: any): ResourceProps[] {
```
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/topology/TopologyResourcePanel.tsx]: nit: can we add a type here?

File: frontend/packages/dev-console/src/components/topology/shapes/DefaultGroup.tsx [hunk 2]
```diff
@@ -82,17 +82,27 @@ class DefaultGroup extends React.Component<GroupProps, DefaultGroupState>
...
-    const { name, dropTarget } = this.props;
+    const { name, dropTarget, selected, onSelect } = this.props;
     const { nodes, lowestPoint, containerPath } = this.state;
 
     if (nodes.length === 0) {
       return null;
     }
 
-    const pathClasses = classNames('odc-default-group', { 'is-highlight': dropTarget });
+    const handleClick = (e) => {
```
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/topology/shapes/DefaultGroup.tsx]: nit: can you add event type here?

**Comment by divyanshiGupta** [comment]
@jeff-phillips-18 for this story are we only going to show the name of the application group in the sidebar? Also the title in the design doc has a resource badge in front of the name and the title color is also different.

This is how it looks right now
Screenshot from 2019-09-25 21-40-34


Design doc
Screenshot from 2019-09-25 21-40-05

Also, there is a blue border around the group. Are we going to add that ?

**Comment by jeff-phillips-18** [comment]
I am updating the styles per the design doc now. I had a few other questions on some of the styling as well.

I don't believe we have the application badge yet but I will confirm.

**Comment by jeff-phillips-18** [comment]
Added styling and application badge.

**Comment by divyanshiGupta** [comment]
@jeff-phillips-18 Code looks good to me. Also I tried it locally, works fine.

/lgtm

**Comment by divyanshiGupta** [comment]
@jeff-phillips-18 noticed one thing that the color of the title is still black and not blue as presented in the design doc. Are we going to keep it as black?

**Comment by jeff-phillips-18** [comment]
@divyanshiGupta It is blue in the doc because eventually it should be a link (when applications are actual objects with a list page and such). That doesn't exist yet so it should be black until it becomes a link.

**Comment by divyanshiGupta** [comment]
> @divyanshiGupta It is blue in the doc because eventually it should be a link (when applications are actual objects with a list page and such). That doesn't exist yet so it should be black until it becomes a link.

Yes, agreed.

**Comment by jeff-phillips-18** [comment]
application-selection

@serenamarie125 @Veethika

**Comment by serenamarie125** [comment]
@jeff-phillips-18 questions on badge.  Is there a reason we are using APP rather than A?  And how did you pick the color?  It probably should be a unique badge color.

**Comment by jeff-phillips-18** [comment]
The default for badges is the first 4 characters of the kind. I can update but I chose to start with the default. The color is unique, I chose one close to what was in the mocks that was a  PF3 pallet color (since all the badges are PF3 pallet colors).

**Comment by serenamarie125** [comment]
ok, that's great about the unique color, thanks @jeff-phillips-18 !

**Comment by serenamarie125** [comment]
I'd suggest just utilizing "A" for Application unless others object @jeff-phillips-18

**Comment by jeff-phillips-18** [comment]
Updated to use `A` as the badge.

**Comment by spadgett** [comment]
/hold
for #2865 (sorry)

**Pull request merged**